src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/BaseComponent.cs

```diff
@@ -0,0 +1,189 @@
+        /// Overrides this method if want to handle behavior to restore the previous status.
+        /// </summary>
+        /// <param name="c">Contents</param>
+        /// <since_tizen> 6 </since_tizen>
+        public virtual void OnRestoreContents(Bundle c)
+        {
+        }
+
+        /// <summary>
+        /// Overrides this method if want to handle behavior to save current status.
+        /// </summary>
+        /// <param name="c">Contents</param>
+        /// <since_tizen> 6 </since_tizen>
+        public virtual void OnSaveContent(Bundle c)
+        {
+        }
+
+        internal void OnLanguageChangedCallback(string language)
+        {
+            LocaleChanged?.Invoke(this, new LocaleChangedEventArgs(language));
+        }
+
+        internal void OnDeviceOrientationChangedCallback(int orientation)
+        {
+            DeviceOrientationChanged?.Invoke(this, new DeviceOrientationEventArgs((DeviceOrientation)orientation));
+        }
+
+        internal void OnLowBatteryCallback(int status)
+        {
+            LowBattery?.Invoke(this, new LowBatteryEventArgs((LowBatteryStatus)status));
+        }
+
+        internal void OnLowMemoryCallback(int status)
+        {
+            LowMemory?.Invoke(this, new LowMemoryEventArgs((LowMemoryStatus)status));
+        }
+
+        internal void OnRegionFormatChangedCallback(string region)
+        {
+            RegionFormatChanged?.Invoke(this, new RegionFormatChangedEventArgs(region));
+        }
+
+        internal void OnSuspendedStateCallback(int state)
+        {
+            SuspendedStateChanged?.Invoke(this, new SuspendedStateEventArgs((SuspendedState)state));
+        }
+
+        /// <summary>
+        /// Sends the launch request asynchronously.
+        /// </summary>
+        /// <remarks>
+        /// To use group mode, you must use this function instead of SendLaunchRequestAsync().
+        /// </remarks>
+        /// <param name="control">appcontrol object</param>
+        /// <param name="replyAfterLaunching">The callback function to be called when the reply is delivered.</param>
+        /// <returns>A task with the result of the launch request.</returns>
+        /// <exception cref="ArgumentException">Thrown when failed because of the argument is invalid.</exception>
+        /// <exception cref="Exceptions.AppNotFoundException">Thrown when the application to run is not found.</exception>
+        /// <exception cref="Exceptions.LaunchRejectedException">Thrown when the launch request is rejected.</exception>
+        /// <privilege>http://tizen.org/privilege/appmanager.launch</privilege>
+        /// <since_tizen> 6 </since_tizen>
+        public Task<AppControlResult> SendLaunchRequestAsync(AppControl control, AppControlReplyCallback replyAfterLaunching)
+        {
+            int ret = Interop.AppControl.SetCallerInstanceId(control.SafeAppControlHandle, Id);
+            if (ret != 0)
+                throw new ArgumentException("Failed to set id");
```

Comment:
Not correct exception, id is not argument on this api

Reply:
Changed it into InvalidOperationException